docs: add more on transaction validity (#597)
omisego/plasma-contracts: plasma_framework/docs/integration-docs/integration-doc.md

## Transaction type, output type, and spending condition

ALD introduces the concepts of transaction type and transaction output type.

> Each transaction output type can define its own rule about how funds may be spent.

Reviewer: `its own set of rules`

And then I'd add more explanation of the concept of a spending condition, before diving into links to code and "gory details" like the `hash` part. Like what is the abstraction, that a spending condition is defined per the pair `tx_type` `output_type` and so forth.

Author: I've provided an example of how spending condition can be used, before the explanation of how it is implemented.

Reviewer: hm, the example is nice (consider choosing sth less speculative though, like a EC signature, which you can then link to our existing Payment tx spending condition). Also mind to keep it to a separate paragraph (check rendered `.md`)

However, I'd insist on providing some not-an-example explanation of a spending condition concept. Just explain what it is, in 2-3 sentences, one of them could literally be "a spending condition is defined per the pair of a transaction type and an output type".

Rules are encoded in the [mapping](../contracts/src/exits/registries/SpendingConditionRegistry.sol) from `hash(tx_type, output_type)` to an address.
If address is 0x0, spend is forbidden.

> Otherwise, address is referencing a contract that implements [ISpendingCondition](../contracts/src/exits/interfaces/ISpendingCondition.sol) interface and its `verify(...) returns (bool)` function.

Reviewer: reference what we have for our current payment transaction, like you're doing for `IStateTransitionVerification`

Author: Done

For a valid transaction, `verify(...)` will return `true` for every of transaction's inputs.

> ## Printing money

Reviewer: I'd opt for a less jargon-ey title :thinking: "Correct transaction balancing", "Correct Input/Output balance", "Transaction balance validation"?

And also in general, so that it is "positively phrased", so "Avoiding printing money" at the minimum.

Author: `Transaction balance validation` it is.

A signature is a specific form of a `witness` and is called as such throughout this document.

## Payment transaction format
Payment transactions are used to transfer fungible tokens, such as ETH and ERC20 tokens. A Payment transaction's output is as described in [FungibleTokenOutputModel](../contracts/FungibleTokenOutputModel.md).

[PaymentTransactionModel](../contracts/PaymentTransactionModel.md) does few more checks when decoding comparing to GenericTransaction above:

> * Is amount of inputs too large? See `_MAX_INPUT_NUM`.  

Reviewer 2: ```suggestion
* Is the number of inputs too large? See `_MAX_INPUT_NUM`.  
```

> * Is outputs number in range `(0, _MAX_OUTPUT_NUM]`?  

Reviewer 2: ```suggestion
* Is the number outputs in the range `(0, _MAX_OUTPUT_NUM]`?  
```

> * None of input pointers is null.

Reviewer 2: ```suggestion
* None of the input pointers is null.
```
